tests/utils/test_list_gsoc.py

```diff
@@ -0,0 +1,124 @@
+        code is located atÂ https://github.com/EOL</p>"""
+            ),
+        )
+
+    def test_get_link_match_repo(self):
+        self.assertEqual(
+            ["beamcommunity/beamcommunity.github.com"],
+            get_link_matches(
+                """https://github.com/beamcommunity/beamcommunity.github.com/wiki"""
+            ),
+        )
+
+    def test_get_link_match_gist(self):
+        # github gists should not match
+        self.assertEqual(
+            [],
+            get_link_matches(
+                "https://gist.github.com/sniok/fb6a1bcf4cf78303a874fa2389739764"
+            ),
+        )
+
+    def test_get_early_archive_project_ext_link(self):
+        fragment = """
+        <span class="mdl-list__item-primary-content">
+        <span class="small-logo-box mdl-list__item-icon">
+        <i class="material-icons small-logo">business</i>
+        </span>
+        <a href="/archive/gsoc/2015/orgs/beamcommunity">BEAM Community</a>
+        </span>
+                """
+        soup = bs4.BeautifulSoup(fragment, features="html.parser")
+        self.assertEqual(
+            {
+                "name": "BEAM Community",
+                "link": "https://www.google-melange.com/archive/gsoc/2015/orgs/beamcommunity",
+                "repos": ["beamcommunity/beamcommunity.github.com"],
+                "year": 2016,
+            },
+            get_early_archive_project(soup, 2016),
+        )
+
+    def test_get_early_archive_project_body_link(self):
+        fragment = """
+        <span class="mdl-list__item-primary-content">
+        <span class="small-logo-box mdl-list__item-icon">
+        <img alt="Encyclopedia of Life logo" class="small-logo" src="/archive/gsoc/2015/orgs/eol/logo-30.png"/>
+        </span>
+        <a href="/archive/gsoc/2015/orgs/eol">Encyclopedia of Life</a>
+        </span>
+        """
+        soup = bs4.BeautifulSoup(fragment, features="html.parser")
+        self.assertEqual(
+            {
+                "name": "Encyclopedia of Life",
+                "link": "https://www.google-melange.com/archive/gsoc/2015/orgs/eol",
+                "repos": ["EOL"],
+                "year": 2015,
+            },
+            get_early_archive_project(soup, 2015),
+        )
+
+    def test_get_early_archive_project_student_project_link(self):
+        fragment = """
+        <span class="mdl-list__item-primary-content">
+        <span class="small-logo-box mdl-list__item-icon">
+        <img alt="Mono Project logo" class="small-logo" src="/archive/gsoc/2015/orgs/mono/logo-30.png"/>
+        </span>
+        <a href="/archive/gsoc/2015/orgs/mono">Mono Project</a>
+        </span>
+        """
+        soup = bs4.BeautifulSoup(fragment, features="html.parser")
+        self.assertEqual(
+            {
+                "name": "Mono Project",
+                "link": "https://www.google-melange.com/archive/gsoc/2015/orgs/mono",
+                "repos": ["ddobrev/QtSharp"],
+                "year": 2017,
```

Comment:
I bet I'm missing something. Why is the returned `year` value `2017` but line 91 and 93 (inside the `fragment` definition) refers to 2015. That's confusing to me.

Reply:
Ah yeah that's confusing - get_early_archive_project takes the year from its second input param rather than parsing the fragment. That said, the mismatch is because this is test data I made up and I forgot to update the year in the fragment. I've done that so it won't confuse anyone else, thanks for catching!